include/vkwrap/vkwrap/utils.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+
+#include "vkwrap/core.h"
+
+#include "utils/algorithm.h"
+
+#include <range/v3/algorithm/contains.hpp>
+#include <range/v3/range/concepts.hpp>
+#include <range/v3/view.hpp>
+
+namespace vkwrap
+{
+
+class SharingInfoSetter
+{
+
+  private:
+    std::vector<QueueFamilyIndex> m_unique_indices;
+    vk::SharingMode m_mode;
+
+  public:
+    SharingInfoSetter( ranges::range auto&& queues )
+    {
+        // clang-format off
+        auto indices = ranges::views::transform( queues, []( auto&& queue ) { 
+            return queue.familyIndex(); 
+        });
+        // clang-format on
+
+        m_unique_indices = utils::getUniqueElements( indices );
+
+        if ( m_unique_indices.size() >= 2 )
+        {
+            m_mode = vk::SharingMode::eConcurrent;
+        } else
+        {
+            m_mode = vk::SharingMode::eExclusive;
+        }
+    } // SharingInfoSetter
+
+    void setTo( auto& create_info ) const&
+    {
+        create_info.setSharingMode( m_mode );
+        create_info.setQueueFamilyIndexCount( static_cast<uint32_t>( m_unique_indices.size() ) );
+        create_info.setPQueueFamilyIndices( m_unique_indices.data() );
+    } // setTo
+
```

Comment:
Why is this a template function?

Reply:
Because there vk::ImageCreateInfo, vk::BufferCreateInfo and etc.